premium/offline.go

```diff
@@ -41,20 +48,83 @@ var publicKey string
 
 var timeFunc = time.Now
 
-func ValidateLicense(logger zerolog.Logger, meta plugin.Meta, licenseFileOrDirectory string) error {
-	fi, err := os.Stat(licenseFileOrDirectory)
+//go:generate mockgen -package=mocks -destination=../premium/mocks/licensemanager.go -source=offline.go AWSLicenseManagerInterface
+type AWSLicenseManagerInterface interface {
+	CheckoutLicense(ctx context.Context, params *licensemanager.CheckoutLicenseInput, optFns ...func(*licensemanager.Options)) (*licensemanager.CheckoutLicenseOutput, error)
+}
+
+type CQLicenseClient struct {
+	logger                  zerolog.Logger
+	meta                    plugin.Meta
+	licenseFileOrDirectory  string
+	awsLicenseManagerClient AWSLicenseManagerInterface
+}
+
+type LicenseClientOptions func(updater *CQLicenseClient)
+
+func WithMeta(meta plugin.Meta) LicenseClientOptions {
+	return func(cl *CQLicenseClient) {
+		cl.meta = meta
+	}
+}
+
+func WithLicenseFileOrDirectory(licenseFileOrDirectory string) LicenseClientOptions {
+	return func(cl *CQLicenseClient) {
+		cl.licenseFileOrDirectory = licenseFileOrDirectory
+	}
+}
+
+func WithAWSLicenseManagerClient(awsLicenseManagerClient AWSLicenseManagerInterface) LicenseClientOptions {
+	return func(cl *CQLicenseClient) {
+		cl.awsLicenseManagerClient = awsLicenseManagerClient
+	}
+}
+
+func NewLicenseClient(ctx context.Context, logger zerolog.Logger, ops ...LicenseClientOptions) (CQLicenseClient, error) {
+	cl := &CQLicenseClient{
```

Comment:
Why create a ptr and then dereference it during return? do `cl := CQLicenseClient{...` and then `op(&cl)` instead?

Reply:
Implemented in [2d9b605](https://github.com/cloudquery/plugin-sdk/pull/1889/commits/2d9b6050b54c775e0137cb154701e2a8ea027a22)